content/articles/flow-and-typescript-2.md

```diff
@@ -0,0 +1,108 @@
+Title: Flow and TypeScript part 2: TypeScript
+Date: 2015-11-20
+Short_summary: Static typing helps keeping codebases (and developers) sane. We have a look on adding types to JavaScript
+Category: Dev
+Authors: Vincent, Tom
+
+
+Continuing from the [first part that introduced Flow](https://blog.wearewizards.io/flow-and-typescript-part-1-flow), we are now going to have a look at [TypeScript](http://www.typescriptlang.org/).
+
+TypeScript is a superset of JavaScript that compiles to several targets, including ES5 and ES6. The current latest version of TypeScript (1.6 at the time of writing) now supports JSX, which was the main blocker for most people using React.
+
+One of the great advantages of TypeScript is  [DefinitelyTyped](https://github.com/borisyankov/DefinitelyTyped). This repo contains definition for over a thousand JavaScript packages but more on that later.
+
+Here is the [repo](https://github.com/Keats/flow-typescript/tree/master/typescript). Note that I have used TypeScript for a project before.
+
+
+## Converting a project to Typescript
+Moving a project to TypeScript is not as straightforward as it was for Flow.
+
+There are a couple of steps and gotchas which I will explain on the way.
+
+### Build tools
+Obviously you are going to need to change your build tools to use TypeScript rather than what you were using before.
+In my case, I was using Webpack so I only had to replace `eslint` with `tslint` and add [ts-loader](https://github.com/TypeStrong/ts-loader) to the mix.
+
+As mentioned in the introduction, TypeScript can target ES5 or ES6. The advantage of targeting ES6 is that you can still use Babel to compile and benefit from hot module reload. For more information on how to do that, please read [ES6 modules with TypeScript and webpack](www.jbrantly.com/es6-modules-with-typescript-and-webpack/) and have a look at the [webpack config](https://github.com/Keats/flow-typescript/blob/master/typescript/webpack.config.js) for a practical example.
+
+
+## Actual conversion
+TypeScript use a config file named `tsconfig.json`. Here's the one currently used by the example:
+
+```ts
+{
+  "compilerOptions": {
+    "target": "es6",
+    "jsx": "react",
+    "noImplicitAny": true,
+    "sourceMap": true
+  },
+  "exclude": ["node_modules", "gulpfile.js", "webpack.config.js", "server.js"]
+}
+```
+When you are starting the conversion, you want to remove the `"noImplicitAny": true,` line since it would error if you have any missing type annotations.
+
+You then need to rename all your `.js` files to `.ts` and `.tsx`. The `.txs` extension is needed if the file contains JSX.
+
+## Some gotchas and examples
+While TypeScript intends to support most of ES6 and ES7 features, you will need to change some bits of your code. Here are the main ones I stumbled into.
```

Comment:
"stumbled into" is the wrong idiom I think, "I ran into" or "I found / discovered" is better.

stumbling into a situation is "blundering into"